TypeError: 'float' object is not callable

How to reproduce:
(1) clone [https://github.com/googleapis/python-cloudbuild](https://github.com/googleapis/python-cloudbuild) and checkout `sijun` branch.
(2) run `python -m nox -s unit-3.6 -- -s`

Exception thrown:
```
    def _apply_decorators(func, decorators):
        """Apply a list of decorators to a given function.
    
        ``decorators`` may contain items that are ``None`` or ``False`` which will
        be ignored.
        """
        decorators = filter(_is_not_none_or_false, reversed(decorators))
    
        for decorator in decorators:
>           func = decorator(func)
E           TypeError: 'float' object is not callable
```

Description of the bug:

`google.api_core.gapic_v1.method._GapicCallable.__call__` [calls](https://github.com/googleapis/python-api-core/blob/master/google/api_core/gapic_v1/method.py#L115) `_determine_timeout` to create an `timeout_` object, then passes it to `_apply_decorators(self._target, [retry, timeout_])`. 

In my example, `_determine_timeout(600.0, 600.0, None)` creates/returns `600.0`, which is a float object, when it is passed to `_apply_decorators`, the 'float' object is not callable is raised. 

`_determine_timeout` should always return a Timeout object.

--- a/.gitignore
+++ b/.gitignore
@@ -29,6 +29,7 @@ pip-log.txt
 .nox
 .cache
 .pytest_cache
+pytype_output
 
 
 # Mac

--- a/google/api_core/gapic_v1/method.py
+++ b/google/api_core/gapic_v1/method.py
@@ -61,6 +61,13 @@ def _determine_timeout(default_timeout, specified_timeout, retry):
     Returns:
         Optional[Timeout]: The timeout to apply to the method or ``None``.
     """
+    # If timeout is specified as a number instead of a Timeout instance,
+    # convert it to a ConstantTimeout.
+    if isinstance(specified_timeout, (int, float)):
+        specified_timeout = timeout.ConstantTimeout(specified_timeout)
+    if isinstance(default_timeout, (int, float)):
+        default_timeout = timeout.ConstantTimeout(default_timeout)
+
     if specified_timeout is DEFAULT:
         specified_timeout = default_timeout
 
@@ -78,12 +85,7 @@ def _determine_timeout(default_timeout, specified_timeout, retry):
         else:
             return default_timeout
 
-    # If timeout is specified as a number instead of a Timeout instance,
-    # convert it to a ConstantTimeout.
-    if isinstance(specified_timeout, (int, float)):
-        return timeout.ConstantTimeout(specified_timeout)
-    else:
-        return specified_timeout
+    return specified_timeout
 
 
 class _GapicCallable(object):

--- a/tests/unit/gapic/test_method.py
+++ b/tests/unit/gapic/test_method.py
@@ -32,6 +32,27 @@ def _utcnow_monotonic():
         curr_value += delta
 
 
+def test__determine_timeout():
+    # Check _determine_timeout always returns a Timeout object.
+    timeout_type_timeout = timeout.ConstantTimeout(600.0)
+    returned_timeout = google.api_core.gapic_v1.method._determine_timeout(
+        600.0, 600.0, None
+    )
+    assert isinstance(returned_timeout, timeout.ConstantTimeout)
+    returned_timeout = google.api_core.gapic_v1.method._determine_timeout(
+        600.0, timeout_type_timeout, None
+    )
+    assert isinstance(returned_timeout, timeout.ConstantTimeout)
+    returned_timeout = google.api_core.gapic_v1.method._determine_timeout(
+        timeout_type_timeout, 600.0, None
+    )
+    assert isinstance(returned_timeout, timeout.ConstantTimeout)
+    returned_timeout = google.api_core.gapic_v1.method._determine_timeout(
+        timeout_type_timeout, timeout_type_timeout, None
+    )
+    assert isinstance(returned_timeout, timeout.ConstantTimeout)
+
+
 def test_wrap_method_basic():
     method = mock.Mock(spec=["__call__"], return_value=42)
 
@@ -154,7 +175,8 @@ def test_wrap_method_with_default_retry_and_timeout_using_sentinel(unusued_sleep
 
 @mock.patch("time.sleep")
 def test_wrap_method_with_overriding_retry_and_timeout(unusued_sleep):
-    method = mock.Mock(spec=["__call__"], side_effect=[exceptions.NotFound(None), 42])
+    method = mock.Mock(spec=["__call__"], side_effect=[
+                       exceptions.NotFound(None), 42])
     default_retry = retry.Retry()
     default_timeout = timeout.ConstantTimeout(60)
     wrapped_method = google.api_core.gapic_v1.method.wrap_method(

